images/runtime/dotnetcore/7.0/base.buster.staging.Dockerfile

```diff
@@ -0,0 +1,67 @@
+# syntax=docker/dockerfile:1.3
+# the above line allows this dockerfile to use the secrets fucntionality
+
+# dotnet tools are currently available as part of SDK so we need to create them in an sdk image
+# and copy them to our final runtime image
+FROM mcr.microsoft.com/dotnet/sdk:7.0 AS tools-install
+RUN dotnet tool install --tool-path /dotnetcore-tools dotnet-sos
+RUN dotnet tool install --tool-path /dotnetcore-tools dotnet-trace
+RUN dotnet tool install --tool-path /dotnetcore-tools dotnet-dump
+RUN dotnet tool install --tool-path /dotnetcore-tools dotnet-counters
+RUN dotnet tool install --tool-path /dotnetcore-tools dotnet-gcdump
+RUN dotnet tool install --tool-path /dotnetcore-tools dotnet-monitor --version 6.*
+
+FROM mcr.microsoft.com/mirror/docker/library/debian:buster-slim
+ARG BUILD_DIR=/tmp/oryx/build
+ADD build ${BUILD_DIR}
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+        # .NET Core dependencies
+        libc6 \
+        libgcc1 \
+        libgssapi-krb5-2 \
+        libicu63 \
+        libssl1.1 \
+        libstdc++6 \
+        zlib1g \
+        lldb \
+        curl \
+        file \
+        libgdiplus \
+    && apt-get upgrade --assume-yes \
+    && rm -rf /var/lib/apt/lists/*
+
+# Configure web servers to bind to port 80 when present
+ENV ASPNETCORE_URLS=http://+:80 \
+    # Enable detection of running in a container
+    DOTNET_RUNNING_IN_CONTAINER=true \
+    PATH="/opt/dotnetcore-tools:${PATH}"
+
+COPY --from=tools-install /dotnetcore-tools /opt/dotnetcore-tools
+
+# Install .NET Core
+# mount the secret sas token to pull the binaries, and make sure we do not print to docker build logs
+RUN --mount=type=secret,id=DOTNET_PRIVATE_STORAGE_ACCOUNT_ACCESS_TOKEN \
+    set -e \
+    # based on resolution on https://github.com/NuGet/Announcements/issues/49#issue-795386700
+    && apt-get remove ca-certificates -y \
+    && apt-get purge ca-certificates -y \
+    && apt-get update \
+    && apt-get install -f ca-certificates=20200601~deb10u2 -y --no-install-recommends \
+    && . ${BUILD_DIR}/__dotNetCoreRunTimeVersions.sh \
+    && curl -SL --output dotnet.tar.gz https://dotnetcli.blob.core.windows.net/dotnet-private/internal/$NET_CORE_APP_70/dotnet-runtime-$NET_CORE_APP_70-linux-x64.tar.gz$(cat /run/secrets/DOTNET_PRIVATE_STORAGE_ACCOUNT_ACCESS_TOKEN) \
```

Comment:
I originally had code that would rewrite base.buster.Dockerfile based on whether the version was to be staged or not, but honestly there were enough changes like comments and things that this ended up being a more straightforward way to implement. In the future I think we could try using templates or something to swap values of a single file, but with the time crunch for this, I didn't think the added complexity was worth it.